src/compiler/factoryPublic.ts

```diff
@@ -2270,16 +2270,13 @@ namespace ts {
             : node;
     }
 
-    export function createDynamicImport(argumentsArray: readonly Expression[], typeArguments: readonly TypeNode[] = []) {
-        const importToken: any = createToken(SyntaxKind.ImportKeyword);
-        return createCall(importToken, typeArguments, argumentsArray);
+    export function createDynamicImportExpression(importKeyword: Token<SyntaxKind.ImportKeyword> | undefined, argumentsArray: readonly Expression[], typeArguments: readonly TypeNode[] = []) {
+        importKeyword = importKeyword || createToken(SyntaxKind.ImportKeyword);
+        return <ImportCall>createCall(<ImportExpression>importKeyword, typeArguments, argumentsArray);
     }
 
-    export function updateDynamicImport(node: CallExpression, argumentsArray: readonly Expression[], typeArguments: readonly TypeNode[] = []) {
-        if (!isDynamicImport(node)) {
-            Debug.fail("updateDynamicImport called with a non dynamic import node.");
-        }
-        return updateCall(node, node.expression, typeArguments, argumentsArray);
+    export function updateDynamicImportExpression(node: ImportCall, importKeyword: ImportExpression, argumentsArray: readonly Expression[], typeArguments: readonly TypeNode[] = []) {

```

Comment:
As above, please ensure the arguments match the order of the nodes as they appear in the syntax, and do not provide a default value for `typeArguments`.
```suggestion
    export function updateDynamicImportExpression(node: ImportCall, importKeyword: ImportExpression, typeArguments: readonly TypeNode[] | undefined, argumentsArray: readonly Expression[]) {
```